File: agent-driven-architecture-main/examples/typescript/test/gate/fixtures/violating/C4/src/spine/pure/staged.ts
// VIOLATION: 11.2 — a staged input that could CARRY authority. The whole claim
// is "recall confers no authority BY CONSTRUCTION": the field's absence IS the
// guarantee, so the field's presence is the violation.
import type { Authority } from "../../spine/pure/actor";

export interface Recalled {
  readonly kind: "Recalled";
  readonly text: string;
  readonly authority: Authority;
}

File: agent-driven-architecture-main/examples/kotlin/src/test/fixtures/konsist/violating/C4/spine/pure/Staged.kt
// BLOCK-TEST C4(e) — a staged input that could CARRY authority.
// 11.2's whole claim is "recall confers no authority BY CONSTRUCTION": the
// field's absence IS the guarantee, so the field's presence is the violation.
// An injected relay entry demanding an irreversible confirmation is refused at
// the gate precisely because a Recalled has nothing that could answer it.
package adr.spine.pure

sealed class StagedInput(open val source: SourceName) {
    data class Recalled(
        override val source: SourceName,
        val text: String,
        val authority: Authority,
    ) : StagedInput(source)
}
